messages/api.go

```diff
@@ -78,8 +68,8 @@ type EndpointMetric struct {
 
 	// This field is true if this distribution is not cumulative.
 	IsNotCumulative bool `json:"isNotCumulative,omitempty"`
-	// The buckets for the distribution.
-	Buckets []Range `json:"buckets,omitempty"`
+	// The upper limits for the distribution buckets.
+	UpperLimits []float64 `json:"upperLimits,omitempty"`
 }
```

Comment:
Also the last bucket never has an upper limit. So +Inf always goes in the last bucket.